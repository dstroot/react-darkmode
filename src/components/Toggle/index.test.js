import React from "react";
import { render } from "@testing-library/react";

// component to test
import Toggle from ".";

// mock toggle function
const toggle = jest.fn();

describe("Toggle", () => {
  it("renders a toggle switch - OFF", () => {
    const { asFragment } = render(
      <Toggle isOn={false} handleToggle={toggle} onColor="blue" />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders a toggle switch - ON", () => {
    const { asFragment } = render(
      <Toggle isOn={true} handleToggle={toggle} onColor="blue" />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
