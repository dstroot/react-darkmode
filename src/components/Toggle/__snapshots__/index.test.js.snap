// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toggle renders a toggle switch - OFF 1`] = `
<DocumentFragment>
  <input
    class="react-switch-checkbox"
    id="react-switch-new"
    type="checkbox"
  />
  <label
    class="react-switch-label"
    for="react-switch-new"
  >
    <span
      class="react-switch-button"
    />
  </label>
</DocumentFragment>
`;

exports[`Toggle renders a toggle switch - ON 1`] = `
<DocumentFragment>
  <input
    checked=""
    class="react-switch-checkbox"
    id="react-switch-new"
    type="checkbox"
  />
  <label
    class="react-switch-label"
    for="react-switch-new"
    style="background: blue;"
  >
    <span
      class="react-switch-button"
    />
  </label>
</DocumentFragment>
`;
